community/unsafe/src/main/java/org/neo4j/unsafe/impl/internal/dragons/UnsafeUtil.java

```diff
@@ -896,4 +921,113 @@ public static boolean exchangeNativeAccessCheckEnabled( boolean newSetting )
         nativeAccessCheckEnabled = newSetting;
         return previousSetting;
     }
+
+    /**
+     * Gets a {@code short} at memory address {@code p} by reading byte for byte, instead of the whole value
+     * in one go. This can be useful, even necessary in some scenarios where {@link #allowUnalignedMemoryAccess}
+     * is {@code false} and {@code p} isn't aligned properly. Values read with this method should have been
+     * previously put using {@link #putShortByteWise(long, short)}.
+     *
+     * @param p address pointer to start reading at.
+     * @return the read value, which was read byte for byte.
+     */
+    public static short getShortByteWise( long p )
```

Comment:
Actually I'd like these method names to also say what endianness they are using. It's good to be explicit about that in case they end up being used together with the non-byte-wise accesses.